### DISCUSSION TITLE
Respect git config status.showUntrackedFile = all

### DISCUSSION THREAD

**Discussion by strindberg** [magit/magit#5259] [Ideas]
Versions: Magit 20241121.944, Transient 20241115.2034, Git 2.47.0, Emacs 29.4, darwin

I have set `magit-section-initial-visibility-alist` to show untracked files:

```
(setq magit-section-initial-visibility-alist (quote ((untracked . show) (stashes . hide))))
```

I also have the following in my .gitconfig:

```
[status]
    showUntrackedFiles = all
```

However, when `magit-status` is invoked, only directories containing untracked files are shown, not the full expanded tree with the individual files. It can of course be expanded manually.

I would expect Magit to respect the git setting and show the individual files right from the start.

**Comment by tarsius** [comment]
This behavior is documented, see `magit-insert-untracked-files`.

The reason `all` is interpreted as "allow showing all, but initially display them collapsed" is that truly displaying all untracked files up front can be very costly and performance is the number one complaint about Magit (even when it is caused by user settings (or choice of OS for that matter)).

The three possible values for `status.showUntrackedFiles` do not unambiguously match to the four possible values that would make sense for Magit, so a choice had to be made. I chose "possibly inconvenient" over "many users would report performance issues".

But it keeps coming up, and I've started work on an (emacs) option that would give the user the four choices (plus "respect the git variable"), but I have stashed this for now.

**Comment by strindberg** [comment]
OK, that's great to hear: I would indeed appreciate the possibility of having the tree initially expanded (a would _not_ then complain about the performance).

**Reply by tarsius** [reply]
This is now supported (since v4.2.1). Note that the *global* value of `status.showUntrackedFiles` is ignored; instead you have to set `magit-status-show-untracked-files` (make sure to re-read its updated docstring).
